handler.go

```diff
@@ -60,6 +63,49 @@ func (o Options) ReadOptions(key string, obj interface{}) error {
 	return json.Unmarshal(msg, obj)
 }
 
+// Stream expects an array of json elements and allows to process them sequentially
+// this helps when one needs to parse a large json without having any memory leaks.
+// Returns ErrEmpty on empty key or when there are no more elements.
+// Returns ErrState when the stream has finished/encountered a Decode error.
+func (o Options) Stream(key string) (func(obj interface{}) error, error) {
+	msg := o[key]
+	if len(msg) == 0 {
+		return nil, errors.Wrap(errors.ErrEmpty, "data")
```

Comment:
I think keeping this consistent to the `ReadOptions` function is better, as for us here `ErrEmpty` is basically the same as noop for `ReadOptions`.

As `Stream` is basically a performance improvement for `ReadOptions` I think it's fair. In which case `ErrEmpty` is a better fit, because it wouldn't require separate handling for when key is not found/iterator has finished. As the only usage example of this we have does not care about the presence of the key and it seems like none of the ones that use `ReadOptions` do too - I'd like to keep it that way.